.github/scripts/ext-debian.sh

```diff
@@ -36,14 +37,42 @@ stage_source() {
   pushd .
 
   echo "Preparing source for $PACKAGE_NAME"
-  cd "$PKG_BUILD_DIR/$PACKAGE_NAME" || exit
+  cd "$PKG_BUILD_PATH/$PACKAGE_NAME" || exit
   debian_package_name=$(dpkg-parsechangelog --show-field Source)
   full_version=$(dpkg-parsechangelog --show-field Version)
   debian_version="${full_version%-*}"
-  cd "$PKG_BUILD_DIR" || exit
+  cd "$PKG_BUILD_PATH" || exit
 
   echo "Generating source tarball from git repo."
-  tar --force-local -c -z -v -f  "${debian_package_name}_${debian_version}.orig.tar.gz" --exclude .git\* --exclude /debian "$PACKAGE_NAME"
+  tar --force-local -c -z -v -f  "${debian_package_name}_${debian_version}.orig.tar.gz" --exclude .git\* --exclude debian "$PACKAGE_NAME"
+
+  if [ "$LOCAL_BUILD" == "false" ]; then
+    debian_package_name_indicator="${debian_package_name:0:1}"
+    if [ "${debian_package_name:0:3}" == "lib" ]; then
+      debian_package_name_indicator="${debian_package_name:0:4}"
+    fi
+
+    # try to download the .orig.tar.gz from existing archive, and check if they are identical or not
+    wget -O "${debian_package_name}_${debian_version}-existing.orig.tar.gz" "http://archive.regolith-desktop.com/$DISTRO/$SUITE/pool/main/${debian_package_name_indicator}/${debian_package_name}/${debian_package_name}_${debian_version}.orig.tar.gz" || true
+
+    if [ -s "${debian_package_name}_${debian_version}-existing.orig.tar.gz" ]; then
+      if ! diff <(tar -tvzf "${debian_package_name}_${debian_version}.orig.tar.gz" | awk '{printf "%10s %s\n",$3,$6}' | sort -k 2 | sed 's|\./||') <(tar -tvzf "${debian_package_name}_${debian_version}-existing.orig.tar.gz" | awk '{printf "%10s %s\n",$3,$6}' | sort -k 2 | sed 's|\./||') ; then
```

Comment:
question: this boils down to just determining the equivalence of two tarballs?

Reply:
Exactly! `.orig.tar.gz` files are shared for each package for each version and all the codenames. The code above is trying to make sure if the existing `.orig.tar.gz` has the same content of the HEAD of the repo. 1) if it does, use that 2) if it doesn't use the new one and then rebuild sources (`.dsc` files) that are pointing to this.